src/cache.rs

```diff
@@ -102,26 +100,26 @@ impl Cache {
     }
 
     /// Return the path to the cache directory.
-    pub fn get_cache_dir() -> Result<(PathBuf, PathSource), TealdeerError> {
+    pub fn get_cache_dir() -> Result<(PathBuf, PathSource)> {
         // Allow overriding the cache directory by setting the env variable.
         if let Ok(value) = env::var(CACHE_DIR_ENV_VAR) {
             let path = PathBuf::from(value);
             let (path_exists, path_is_dir) = path
                 .metadata()
                 .map_or((false, false), |md| (true, md.is_dir()));
             if path_exists && !path_is_dir {
-                return Err(CacheError(format!(
-                    "Path specified by ${} is not a directory.",
+                return Err(anyhow!(
+                    "Path specified by ${} is not a directory",
                     CACHE_DIR_ENV_VAR
-                )));
+                ));
```

Comment:
There are the macros `bail!` and `ensure!` that would capture this more precisely:

```rust
ensure!(!path_exists || path_is_dir, "Path specified, by ${} is not a directory", CACHE_DIR_ENV_VAR);
```